sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -653,5 +652,77 @@ def Transport(host, connect_timeout=None, ssl=False, **kwargs):
     Given a few parameters from the Connection constructor,
     select and create a subclass of _AbstractTransport.
     """
-    transport = SSLTransport if ssl else TCPTransport
+    transport_type = kwargs.pop('transport_type')
```

Comment:
same as async

Reply:
yep - transport_type is always passed. Again, it's not publicly exposed

Reply:
hmm I think we could be more intentional about that. It's currently treated as a required parameter in multiple places, but is never described as such. I don't like relying on just because it's currently "always passed in". If that is the case - let's just make it explicit.
I would say make it required in one place - probably the transport seems like a good fit - and actually make it a positional parameter. Then at the Connection layer or the Client layer we implement the correct "default" logic that gets passed in.